src/main/kotlin/cacophony/parser/Parser.kt

```diff
@@ -9,3 +9,7 @@ interface Parser<SymbolType : Enum<SymbolType>> {
         diagnostics: Diagnostics,
     ): ParseTree<SymbolType>
 }
+
+class ParserError(
+    reason: String,
+) : Exception(reason)
```

Comment:
in #65 we have `ParserConstructorErrorException`, so please change this one to `ParserProcessErrorException`

Reply:
Renamed it to `ParsingErrorException` for consistency, but `ParserProcessErrorException` seems artificial